packages/video_player/lib/video_player.dart

```diff
@@ -261,6 +265,37 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   }
 }
 
+class _VideoAppLifeCycleObserver extends WidgetsBindingObserver {
+  bool wasPlayingBeforePause = false;
+  final VideoPlayerController controller;
+
+  _VideoAppLifeCycleObserver(this.controller);
+
+  void initialize() {
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.paused:
+        wasPlayingBeforePause = controller.value.isPlaying;
+        if (controller.value.initialized) controller.pause();
```

Comment:
Please format single-statement if branches consistently.

Worse: if controller is not yet initialized, we lose the pause signal and may start playing while in background.

Reply:
I removed the conditional. The controller should take care of delaying the play message till after initialisation is done.